api/envoy/config/core/v3/protocol.proto

```diff
@@ -652,5 +652,10 @@ message SchemeHeaderTransformation {
   oneof transformation {
     // Overwrite any Scheme header with the contents of this string.
     string scheme_to_overwrite = 1 [(validate.rules).string = {in: "http" in: "https"}];
+
```

Comment:
We are currently discouraging `oneof` fields to allow control planes to be backward-compatible (see https://github.com/envoyproxy/envoy/blob/main/api/STYLE.md).
I suggest adding the field outside of the oneof, and clarify the precedence of the fields.

Reply:
Addressed, but unsure what the precedence should be. Made the original field `scheme_to_overwrite` take precedence since it is more explicit.